spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/EnvBasedClientConfig.java

```diff
@@ -0,0 +1,100 @@
+package org.springframework.cloud.netflix.ribbon;
+
+import static com.netflix.client.config.CommonClientConfigKey.DeploymentContextBasedVipAddresses;
+
+import java.util.concurrent.TimeUnit;
+
+import com.netflix.client.config.CommonClientConfigKey;
+import org.springframework.boot.bind.RelaxedPropertyResolver;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+import com.netflix.client.config.DefaultClientConfigImpl;
+import com.netflix.client.config.IClientConfigKey;
+
+/**
+ * 
+ * {@link com.netflix.client.config.IClientConfig} implementation that depends on Spring
+ * environment for its values instead of using Archaius
+ * 
+ * @author Biju Kunjummen
+ */
+public class EnvBasedClientConfig extends DefaultClientConfigImpl {
+
+	private RelaxedPropertyResolver propertyResolver;
+
+	public static final IClientConfigKey<String> RETRYABLE_STATUS_CODES = new CommonClientConfigKey<String>("retryableStatusCodes") {};
+
+	public EnvBasedClientConfig(ConfigurableEnvironment environment) {
+		super();
+		this.propertyResolver = new RelaxedPropertyResolver(environment);
+	}
+	
+	@Override
+	public void loadProperties(String restClientName) {
+		setClientName(restClientName);
+		loadDefaultValues();
+	}
+
+	@Override
+	public void loadDefaultValues() {
+		super.loadDefaultValues();
+		putDefaultStringProperty(DeploymentContextBasedVipAddresses, this.getClientName());
+		putDefaultStringProperty(RETRYABLE_STATUS_CODES, "");
+	}
+
+	@Override
+	protected void putDefaultIntegerProperty(IClientConfigKey propName,
```

Comment:
Yes, agreed, these are because I am extending my `IClientConfig` from `DefaultClientConfigImpl`, these specific overrides bypass some of the archaius calls in the base class. However if the approach itself looks okay I can try a clean room implementation of IClientConfig that does not depend on DefaultCleintConfiImpl at all